BucketSort.c

```diff
@@ -0,0 +1,181 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include "BucketSort.h"
+#include "QuickSort.h"
+
+#define INFTY 1.0e+6
+#define TOL 1.0e-14
+#define mabs(a) ((a) >= 0 ? (a) : -(a))
+#define sgn(a) ((a) == 0 ? 0 : ((a) > 0  ? 1 : -1 ))
+#define max(a,b) ((a) >= (b) ? (a) : (b))
+#define min(a,b) ((a) <= (b) ? (a) : (b))
+
+void dial_list_init(struct mylist *list,int ind);
```

Comment:
You don't need to forward declare these functions, since they're already declared in "BucketSort.h".

Reply:
Removed such redundant declaration in BucketSort.c and all other files with .hcounterpart.